k256/src/arithmetic/scalar.rs

```diff
@@ -227,6 +238,82 @@ impl Scalar {
     }
 }
 
+impl ff::Field for Scalar {
+    fn random<R: RngCore + ?Sized>(rng: &mut R) -> Self {
+        Scalar::generate_vartime(rng)
+    }
+
+    fn zero() -> Self {
+        Scalar::zero()
+    }
+
+    fn one() -> Self {
+        Scalar::one()
+    }
+
+    fn is_zero(&self) -> bool {
+        self.0.is_zero().into()
+    }
+
+    #[must_use]
+    fn square(&self) -> Self {
+        Scalar::square(self)
+    }
+
+    #[must_use]
+    fn double(&self) -> Self {
+        todo!();
+    }
+
+    fn invert(&self) -> CtOption<Self> {
+        Scalar::invert(self)
+    }
+
+    fn sqrt(&self) -> CtOption<Self> {
+        todo!();
+    }
+}
+
+impl ff::PrimeField for Scalar {
+    type Repr = ElementBytes;
+    type ReprEndianness = byteorder::BigEndian;
+
+    const NUM_BITS: u32 = 256;
+    const CAPACITY: u32 = 256;
+    const S: u32 = 6;
+
+    fn from_repr(repr: ElementBytes) -> Option<Self> {
+        let result = Scalar::from_bytes(&repr);
+
+        // TODO(tarcieri): replace with into conversion when available (see subtle#73)
```

Comment:
I should really open a PR for this rather than copying this comment everywhere 😅

Reply:
Oh hey look, someone else opened one already: https://github.com/dalek-cryptography/subtle/pull/75